nexus/src/external_api/console_api.rs

```diff
@@ -135,6 +139,52 @@ pub async fn spoof_login_form(
     serve_console_index(rqctx.context()).await
 }
 
+/// Redirect to IdP login URL
+//
+// Currently hard-coded to redirect to our own fake login form.
+#[endpoint {
+   method = GET,
+   path = "/login",
+   unpublished = true,
+}]
+pub async fn login_redirect(
+    _rqctx: Arc<RequestContext<Arc<ServerContext>>>,
+) -> Result<Response<Body>, HttpError> {
+    Ok(Response::builder()
+        .status(StatusCode::FOUND)
+        .header(http::header::LOCATION, "/spoof_login") // TODO: placeholder
+        .body("".into())?)
+}
+
+/// Fetch the user associated with the current session
+#[endpoint {
+   method = GET,
+   path = "/session/me",
+}]
+pub async fn session_me(
+    rqctx: Arc<RequestContext<Arc<ServerContext>>>,
+) -> Result<HttpResponseOk<views::User>, HttpError> {
+    let apictx = rqctx.context();
+    let nexus = &apictx.nexus;
+    let handler = async {
+        let opctx = OpContext::for_external_api(&rqctx).await?;
+        // TODO: we don't care about authentication method, as long as they are
+        // authed as _somebody_. We could restrict this to session auth only,
+        // but it's not clear what the advantage would be.
+        match opctx.authn.actor() {
+            Some(actor) => {
+                let user =
```

Comment:
Related to my top-level comment, this endpoint could essentially just:

- make the OpContext
- do an auth check that you're allowed to look at your own data (or something like that)
- construct a view from the details stored in the authn

i.e., it wouldn't do a database call here and it wouldn't generated its own "Unauthenticated" error.

Reply:
I did this in 825a8d3861f384b0a7f13933982242fae836d3a4 minus the auth check. Is the auth check necessary? Would we need to create a special resource and special read permissions for data living in the auth context?

Reply:
The authz check was just an idea to leverage the authz system's error handling for unauthenticated users.